src/app/features/habit/habitSlice.ts

```diff
@@ -0,0 +1,75 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { PayloadAction } from "@reduxjs/toolkit";
+
+export const habit = {
+  id: null,
+  goal: "",
+  schema: "",
+  startDate: null,
+  rewards: [],
+  habitStrength: {
+    strength: 0,
+    history: [],
+  },
+  data: [],
+};
```

Comment:
I think that this is only an initial state habit state. Using and modifying it externally can be quite dangerous so I would rename it and do not export it.